HungryDevelopers.playground/Contents.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+
+class Spoon {
+    var index: Int = 0
+    private let lock = NSLock()
+    
+    func pickUp() {
+        lock.lock()
+    }
+    
+    func putDown() {
+        lock.unlock()
+    }
+}
+
+class Developer {
+    var name: String?
+    var leftSpoon: Spoon?
+    var rightSpoon: Spoon?
+    
+    func think() {
+        let group = DispatchGroup()
+        let avenueQueue = DispatchQueue(label: "Avenue Queue")
+        
+        if let leftIndex = leftSpoon?.index,
+            let rightIndex = rightSpoon?.index {
+            if leftIndex < rightIndex {
+                group.enter()
+                avenueQueue.async {
+                    self.leftSpoon?.pickUp()
+                    print("\(self.name ?? "") picked up left spoon.")
+                    group.leave()
+                }
+                group.wait()
```

Comment:
Since you are entering and then immediately waiting, this is essentially equivalent to making this all synchronous code... In other words, you could just pickup the spoon, no queue, and no dispatch group.